src/AppBundle/Controller/RestaurantController.php

```diff
@@ -602,8 +602,32 @@ public function mapAction(Request $request, SlugifyInterface $slugify)
     /**
      * @Route("/restaurants/suggest", name="restaurants_suggest")
      */
-    public function suggestRestaurantAction() {
-        echo "IT WORKS";
-        exit;
+    public function suggestRestaurantAction(Request $request, ObjectManager $manager) {
+        $pledge = new Pledge();
+
+        $form = $this->createForm(PledgeType::class, $pledge);
+
+        $form->handleRequest($request);
+
+        if($form->isSubmitted() && $form->isValid()) {
+            $pledge->setCreatedAt(new \DateTime());
+            $pledge->setState(1);
+            $pledge->setUser($this->getUser());
+            $manager->persist($pledge);
+            $manager->flush();
+
+            return $this->redirectToRoute('restaurants_suggest_submitted');
+    }
+        return $this->render('@App/restaurant/restaurant_pledge.html.twig', [
+        'formPledge' => $form->createView()
```

Comment:
In Twig, the coding style is to use snake case, so the variable should be `form_pledge`